commit c9276ae7cfecbd1df477ff31116682e72dd6d8f7
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Mar 9 13:28:24 2016 +0000

    serial: Take care starting a hung-up tty's port
    
    tty_port_hangup sets a port's tty field to NULL (holding the port lock),
    but uart_tx_stopped, called from __uart_start (with the port lock),
    uses the tty field without checking for NULL.
    
    Change uart_tx_stopped to treat a NULL tty field as another stopped
    indication.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/include/linux/serial_core.h b/include/linux/serial_core.h
index 297d4fa..00ce6c6 100644
--- a/include/linux/serial_core.h
+++ b/include/linux/serial_core.h
@@ -397,7 +397,7 @@ int uart_resume_port(struct uart_driver *reg, struct uart_port *port);
 static inline int uart_tx_stopped(struct uart_port *port)
 {
 	struct tty_struct *tty = port->state->port.tty;
-	if (tty->stopped || port->hw_stopped)
+	if (!tty || tty->stopped || port->hw_stopped)
 		return 1;
 	return 0;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1202_c9276ae7cfecbd1df477ff31116682e72dd6d8f7.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1202_c9276ae7cfecbd1df477ff31116682e72dd6d8f7.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
